Slim/Networking/Async/HTTP.pm

```diff
@@ -293,7 +293,9 @@ sub _format_request {
 	}
 
 	# XXX until we support chunked encoding, force 1.0
-	$self->socket->http_version('1.0');
+	# $self->socket->http_version('1.0');
+	$self->socket->http_version($self->request->protocol =~ m|HTTP/(\S*)|i);
```

Comment:
Please store the result of the protocol version extraction in a variable and do some sanity checking. Or are you confident enough that the result of the regex would always be something valid (eg. because that check already is done elsewhere)?

Reply:
I can change that but basically either the application code set the request->protocol explicitely and I assumed that if the user enters something stupid, he/she should not expect it to work. Then, if nothing is specified (most of the cases) then the code sets its explicitely above (205)
`	if ( !$self->request->protocol ) {
		$self->request->protocol( 'HTTP/1.0' );
	}`